lib/io/reader.toit

```diff
@@ -309,19 +309,25 @@ abstract class Reader implements old-reader.Reader:
         end := to ? min (start + to) it.size : it.size
         index := it.index-of byte --from=start --to=end
         if index >= 0: return offset + (index - start)
-        if to: to -= it.size - start
+        if to:
+          to -= it.size - start
+          if to <= 0: return -1
         offset += it.size - start
         start = 0
 
+    if to and to <= 0: return -1
```

Comment:
Maybe move this check up before the check for buffered_? It feels like you don't really have to check if there were any elements in buffered_, because you will have checked this after looking in the last buffered segment.

Also, it is a fairly simple check and it just seems nicer to get it out of the way early on.